<a href="https://colab.research.google.com/github/s18621/PAD/blob/main/Zadanie_9/PAD_09.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
PATH ='./drive/MyDrive/Dane/PAD_09/PAD_09_PD.csv'

Przeprowadź test T studenta na danych w pliku PAD_PD_09.csv
- Hipoteza zerowa brzmi: nie ma istotnej różnicy w średnich Annual Income pomiędzy kobietami i mężczyznami. 
- Hipoteza alternatywna brzmi: istnieje istotna różnica (bez wskazania kierunku) w średnich Annual Income pomiędzy kobietami i mężczyznami.


Przyjmij significance level (alpha) na poziomie 0.05. Czy uda się odrzucić hipotezę zerową?


In [20]:
import pandas as pd
import numpy as np
import plotly.graph_objects as go
import scipy.stats as stats

df = pd.read_csv(PATH,sep=";")

income_m = df.loc[df['Gender'] == 'Male']['Annual Income (k$)'].tolist()
income_f = df.loc[df['Gender'] == 'Female']['Annual Income (k$)'].tolist()

mean_income_m = df.loc[df['Gender'] == 'Male']['Annual Income (k$)'].mean()
mean_income_f = df.loc[df['Gender'] == 'Female']['Annual Income (k$)'].mean()

std_income_m = df.loc[df['Gender'] == 'Male']['Annual Income (k$)'].std()
std_income_f = df.loc[df['Gender'] == 'Female']['Annual Income (k$)'].std()

print(f'Male STD: {std_income_m}')
print(f'Female STD: {std_income_f}')

print(f'Male income: {mean_income_m}')
print(f'Female income: {mean_income_f}')


print(stats.ttest_ind(a=income_f, b=income_m, equal_var=True))

Male STD: 26.63837318249413
Female STD: 26.01195151505595
Male income: 62.22727272727273
Female income: 59.25
Ttest_indResult(statistic=-0.795022298602198, pvalue=0.42755249399927864)


#Nie można odrzucić hipotezy zerowej
pvalue < poziomu istotności

- MEN(Mean:62.22 Std:26.63)
- FEMALE(Mean:59.25, Std:26.01)

#Zadanie 2
Korzystając z tych samych danych i biblioteki statmodels stwórz model regresji liniowej, gdzie zmienną zależną jest spending score, a zmiennymi niezależnymi pozostałe zmienne.
Z modelu wyświetl p values, standard dev i coefficients.
Sprawdź czy istnieje korelacja pomiędzy zmiennymi. Pokaż to na wykresie, np. z biblioteki plotly express.
Korzystając z eliminacji wstecznej usuń najmniej istotną zmienną i jeszcze raz wyświetl model.


In [22]:
import statsmodels.api as sm
import statsmodels.formula.api as smf

regress_df = df.copy()
regress_df.columns = ['Gender', 'Age', 'Annual_Income', 'Spending_Score']

model_df = smf.ols(formula='Spending_Score ~ Annual_Income + Age + C(Gender)', data=regress_df).fit()

print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:         Spending_Score   R-squared:                       0.109
Model:                            OLS   Adj. R-squared:                  0.095
Method:                 Least Squares   F-statistic:                     7.960
Date:                Sun, 29 May 2022   Prob (F-statistic):           4.91e-05
Time:                        14:39:39   Log-Likelihood:                -922.05
No. Observations:                 200   AIC:                             1852.
Df Residuals:                     196   BIC:                             1865.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
Intercept            73.9300      6.64

In [25]:
print(f'Pvalues: \n{model_df.pvalues}\n')
print(f'Bse: \n{model_df.bse}\n')
print(f'Params: \n{model_df.params}')

Pvalues: 
Intercept            1.287752e-22
C(Gender)[T.Male]    5.671172e-01
Annual_Income        9.050939e-01
Age                  3.055177e-06
dtype: float64

Bse: 
Intercept            6.642253
C(Gender)[T.Male]    3.511825
Annual_Income        0.066420
Age                  0.124916
dtype: float64

Params: 
Intercept            73.930034
C(Gender)[T.Male]    -2.013234
Annual_Income         0.007929
Age                  -0.600371
dtype: float64


In [26]:
import plotly.express as px
%matplotlib inline

heatmap = df.copy()
md = sm.tools.categorical(df['Gender'], drop=True)
heatmap['Gender'] = md.iloc[:,0]

fg = px.imshow(heatmap.corr(), text_auto=True)
fg.show()

In [27]:
model_df = smf.ols(formula='Spending_Score ~ Annual_Income - Age + C(Gender)', data=regress_df).fit()
print(model_df.summary())

                            OLS Regression Results                            
Dep. Variable:         Spending_Score   R-squared:                       0.004
Model:                            OLS   Adj. R-squared:                 -0.007
Method:                 Least Squares   F-statistic:                    0.3510
Date:                Sun, 29 May 2022   Prob (F-statistic):              0.704
Time:                        14:50:37   Log-Likelihood:                -933.19
No. Observations:                 200   AIC:                             1872.
Df Residuals:                     197   BIC:                             1882.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
Intercept            50.7565      4.81